pkg/bus/bus.go

```diff
@@ -156,6 +159,10 @@ type Event struct {
 }
 
 func (b *Bus) Publish(topic Topic, message ...Message) (Future, error) {
+	if !b.closer.AddRunning() {
+		return nil, nil
+	}
+	defer b.closer.Done()
 	if topic.ID == "" {
```

Comment:
Put them between line 184 and line 185. Each sender is a running task that should be closed gracefully.